src/core/services/layout/layout.js

```diff
@@ -1,319 +1,441 @@
-
-            attributeToClass(null, element);
-
-            // Use for postLink to account for transforms after ng-transclude.
-            return attributeToClass;
+      }
+    };
+  }
+
+  /**
+   * Provide console warning that this layout attribute has been deprecated
+   *
+   */
+  function warnAttrNotSupported(className) {
+    var parts = className.split("-");
+    return ["$log", function($log) {
+      $log.warn(className + "has been deprecated. Please use a `" + parts[0] + "-gt-<xxx>` variant.");
+      return angular.noop;
+    }];
+  }
+
+  /**
+   * For the Layout attribute value, validate or replace with default
+   * fallback value
+   */
+  function validateAttributeValue(className, value, updateFn) {
+    var origValue = value;
+
+    if (!needsInterpolation(value)) {
+      switch (className) {
+        case 'layout'       :
+        case 'layout-sm'    :
+        case 'layout-gt-sm' :
+        case 'layout-md'    :
+        case 'layout-gt-md' :
+        case 'layout-lg'    :
+        case 'layout-gt-lg' :
+          if ( !match(value, ['row', 'column']) ) {
+            value = 'row';  // Default fallback to row
           }
-        };
-      }];
-
-      /**
-       * Add as transformed class selector, then
-       * remove the deprecated attribute selector
-       */
-      function attributeToClass(scope, element) {
-        element.addClass(className);
-
-        if ( $$mdLayout.removeAttributes ) {
-          // After link-phase, remove deprecated layout attribute selector
-          element.removeAttr(className);
-        }
+          break;
+
+        case 'flex'       :
+        case 'flex-sm'    :
+        case 'flex-gt-sm' :
+        case 'flex-md'    :
+        case 'flex-gt-md' :
+        case 'flex-lg'    :
+        case 'flex-gt-lg' :
+          if (!match(value, ['grow', 'initial', 'auto', 'none'])) {
+            if (isNaN(+value)) {
+              value = '';  // Default fallback
+            }
+          }
+          break;
+
+        case 'layout-offset'      :
+        case 'layout-offset-sm'   :
+        case 'layout-offset-gt-sm':
+        case 'layout-offset-md'   :
+        case 'layout-offset-gt-md':
+        case 'layout-offset-lg'   :
+        case 'layout-offset-gt-lg':
+        case 'flex-order'       :
+        case 'flex-order-sm'    :
+        case 'flex-order-gt-sm' :
+        case 'flex-order-md'    :
+        case 'flex-order-gt-md' :
+        case 'flex-order-lg'    :
+        case 'flex-order-gt-lg' :
+
+          if (!value || isNaN(+value)) {
+            // Default fallback to flex == {flex: 1;}
+            value = '0';  // Default fallback
+          }
+          break;
+
+        case 'layout-align'       :
+        case 'layout-align-sm'    :
+        case 'layout-align-gt-sm' :
+        case 'layout-align-md'    :
+        case 'layout-align-gt-md' :
+        case 'layout-align-lg'    :
+        case 'layout-align-gt-lg' :
+          var alignmentValues = [
```

Comment:
I considered that... but since the var and values are only used here (for these cases), hoisting it would disconnect the context.